iroh-resolver/src/unixfs.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::{anyhow, bail, Result};
+use bytes::{Buf, Bytes};
+use cid::Cid;
+use prost::Message;
+
+mod unixfs_pb {
+    include!(concat!(env!("OUT_DIR"), "/unixfs_pb.rs"));
+}
+
+mod dag_pb {
+    include!(concat!(env!("OUT_DIR"), "/merkledag_pb.rs"));
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, num_enum::IntoPrimitive, num_enum::TryFromPrimitive,
+)]
+#[repr(i32)]
+pub enum DataType {
+    Raw = 0,
+    Directory = 1,
+    File = 2,
+    Metadata = 3,
+    Symlink = 4,
+    HamtShard = 5,
+}
```

Comment:
Ok, q. Doesn't some version of this `DataType` enum get automatically generated when we create the unixfs protobuf?

Or is it not as useful because it's nested inside of `data`? Or am I missing something?

Reply:
this matches the protobuf definition and the derive macro for nums generates conversion methods  for the numbers

Reply:
The protobuf generator doesn't actually give us an enum, only constants, so this is not relly a redefiniton